### ISSUE TITLE
Managed Navigation with friendly urls and re-render of the PnP search results webpart

### ISSUE THREAD

**Issue by ileana2019** [microsoft-search/pnp-modern-search#495]
**Is your feature request related to a problem? Please describe.**
When using managed navigation with friendly urls, all pointing to the same results modern page (which contains the PnP search results webpart), and navigating from one navigation item to the other, the PnP Search results webpart is not re-render. The friendly urls are used to filter the search results webpart (example: Query template settings is {searchTerms} RefinableString120:{URLToken.1}). 

**Describe the solution you'd like**
Add a toggle control in the webpart properties, which will re-render the webpart is the query string parameter of the page will change (see [link](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/guidance/intercepting-query-changes-in-webparts)).

**Describe alternatives you've considered**
Create another webpart to be added on the same page as the search results webpart, which will reload the page.

**Comment by wobba** [comment]
Configure your result wp like the below image. You can use any param you like. I just happened to use `s`. Be sure to have the param in the URL when you edit the page to pick it up. Much better approach compared to the query variables. Does this work for you?

![image](https://user-images.githubusercontent.com/51104/94560413-e9f30300-0262-11eb-8ff5-f463d08b2864.png)

**Comment by ileana2019** [comment]
Hello,
Hmm, not really. So, we have this TermSet at the site collection level used for the managed navigation:

![Nav Termset](https://user-images.githubusercontent.com/55545989/94568725-2f1c3280-026d-11eb-84b3-fedda858edd3.png)

Each term of the term set is configured to use friendly urls like this:

![Term config](https://user-images.githubusercontent.com/55545989/94569229-c2edfe80-026d-11eb-982b-948979504809.png)

Where the GUID is the term GUID.

The "Search-results.aspx" page where this friendly url is redirecting, is having the Search Results webpart added to the page. In the webpart properties, inside the Query template we have: RefinableString130:{URLToken.1}, to filter the results coming from a list (items in the list are tagged with the navigation term).

The problem is that when we navigate, via the top navigation (which is a managed navigation), from one term to another (example when clicking in the top navigation Bot then TOP), the search results are still displaying Bot (because the page or the Search results webpart is not re-rendering, even if the url has changed from "/Comp/c64ee714-6638-437a-bdef-8e194b455f31" to "/Comp/b8426391-ea9a-45e3-8abd-9c5b72f8577f"). 
As the link above describes it:
"However if the user proceeds to click a link on the page that contains the same path as the current URL and only the query string changes, the URL in the address bar will be updated but the page won't re-render (because of performance reasons)."

**Comment by wobba** [comment]
Haven't really tested this, but are you using classic publishing pages with this, or does it work with modern pages as well? I'll tag this as a bug in any case to see if we can solve it. @MakingThingsWorkToronto

**Comment by ileana2019** [comment]
It is modern pages only. In order to use the managed navigation, the Publishing feature had to be activated.

**Comment by wobba** [comment]
@ileana2019 Using classic publishing and managed nav... your choice :)

**Comment by wobba** [comment]
@ileana2019  could you retry with the latest dev version?

**Comment by darioca** [comment]
I'm having the same issue. I'm using the menu items in the Mega menu to open the same page and passing different query parameters.  
Example:
/sites/sitename/SitePages/Search.aspx?cat=red
/sites/sitename/SitePages/Search.aspx?cat=blue

These parameters are used to filter the search result (e.g. RefinableString00:{QueryString.cat} ).
The "PnP - Search Results" wp is connected to a "PnP - Search Box" wp on the same page to allow refining the results, and therefore I can't connect it to Page Environment/Query String as suggested.

The problem is that the page is refreshed and the search results rendered only if I click on the same menu link twice.

Is there a workaround to force the page to re-render?
Or can something be changed to allow the search results to recalculate even if we're still on the same page and the querystring has changed?

**Comment by wobba** [comment]
Are you using v3 or v4?

**Comment by darioca** [comment]
V4.1

**Comment by wobba** [comment]
Verified this using the hub navigation.

* Add cat=something to the URL of your page
* Edit the page
* Add search box and result wp and connect them
* Configure search box to pick up the cat parameter 
![image](https://user-images.githubusercontent.com/51104/116701107-d3af5000-a9c7-11eb-915b-e4c175382acf.png)

**Comment by darioca** [comment]
>Are you using an empty search box with a default query for your scenario? Meaning, you want to show results based on the cat param, but search box should be empty when you load initial results?

Yes, correct, I want to show results based on the cat param and an empty search box, so I can use it to refine the initial search results based on the cat parameter.

Example: 
/sites/sitename/SitePages/Search.aspx?cat=car

and then, in the search box, type: "Ford", to display all Ford cars.

**Comment by wobba** [comment]
As a workaround you can pass in '*' for the search box param.. allows for triggering at least.
